examples/grpc/client.cc

```diff
@@ -0,0 +1,109 @@
+#include "grpc_foo_lib/foo_split.h"
+#include "grpc_foo_lib/grpc_map_carrier.h"
+#include "messages.grpc.pb.h"
+#include "tracer_common.h"
+
+#include <grpcpp/grpcpp.h>
+#include <iostream>
+#include <string>
+
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::ClientReader;
+using grpc::Status;
+
+using grpc_example::Greeter;
+using grpc_example::GreetRequest;
+using grpc_example::GreetResponse;
+
+namespace
+{
+class GreeterClient
+{
+public:
+  GreeterClient(std::shared_ptr<Channel> channel) : stub_(Greeter::NewStub(channel)) {}
+
+  std::string Greet(std::string ip, uint16_t port)
+  {
+    // Build gRPC Context objects and protobuf message containers
+    GreetRequest request;
+    GreetResponse response;
+    ClientContext context;
+    request.set_request("Nice to meet you!");
+    // See tracer_common.h for this function
+    auto propagator = get_propagator();
+
+    opentelemetry::trace::StartSpanOptions options;
+    options.kind = opentelemetry::trace::SpanKind::kClient;
+    // Spans are only as useful as the information that you put in them.
+    // Even though this is a client service, it can still be useful to add
+    // as much data as possible. We add attributes for peer ip and port, both
+    // because it is required per OpenTelemetry's rpc semantic conventions,
+    // and because it could still be useful to a debugger in the future.
+    std::string span_name = "GreeterClient/Greet";
+    auto span             = get_tracer("grpc-client")
+                    ->StartSpan(span_name,
+                                {{"rpc.system", "grpc"},
+                                 {"rpc.service", "grpc-example.GreetService"},
+                                 {"rpc.method", "Greet"},
+                                 {"net.peer.ip", ip},
+                                 {"net.peer.port", port},
+                                 {"rpc.grpc.status_code", 0}},
```

Comment:
status-code value should be based on the response received from the server. We should set it after that receiving server response using `span::SetAttribute(key, value)`